toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -79,11 +118,165 @@ func compressWithZstd(partitionRawFilepath string) (partitionFilepath string, er
 		return "", fmt.Errorf("failed to compress %s with zstd:\n%w", partitionRawFilepath, err)
 	}
 
-	// Remove raw file since output partition format is raw-zst.
-	err = os.Remove(partitionRawFilepath)
+	return partitionRawFilepath + ".zst", nil
+}
+
+// Prepend a skippable frame with the metadata payload to the specified partition file.
+func addSkippableFrame(partitionFilepath string, magicNumber uint32, frameSize uint32, payload [4]uint32) (err error) {
+	// Read existing data from the partition file.
+	existingData, err := os.ReadFile(partitionFilepath)
+	if err != nil {
+		return fmt.Errorf("failed to read partition file %s:\n%w", partitionFilepath, err)
+	}
+
+	// Create a skippable frame.
+	skippableFrame := createSkippableFrame(magicNumber, frameSize, payload)
+
+	// Combine the skippable frame and existing data.
+	newData := append(skippableFrame, existingData...)
+
+	// Write the combined data back to the partition file.
+	err = os.WriteFile(partitionFilepath, newData, 0644)
```

Comment:
Where does the mode of 0644 come from? Is this how the file was originally created? If so, can you reuse a constant value here?

Reply:
can you add a comment for this in the file?

Reply:
Added constant and comment.